src/bacnet/basic/bbmd6/h_bbmd6.c

```diff
@@ -612,6 +612,46 @@ static void bbmd6_address_resolution_handler(
     }
 }
 
+/**
+ * Handler for Forwarded-Address-Resolution
+ *
+ * @param addr - BACnet/IPv6 source address any NAK or reply back to.
+ * @param pdu - The received NPDU+APDU buffer.
+ * @param pdu_len - How many bytes in NPDU+APDU buffer.
+ */
+static void bbmd6_forwarded_address_resolution_handler(
+    const BACNET_IP6_ADDRESS *addr, const uint8_t *pdu, uint16_t pdu_len)
+{
+    int function_len = 0;
+    uint32_t vmac_src = 0;
+    uint32_t vmac_target = 0;
+    uint32_t vmac_me = 0;
+    BACNET_IP6_ADDRESS bip6_address = {
+        0,
+    };
+
+    if (addr && pdu) {
+        PRINTF("BIP6: Received Forwarded-Address-Resolution.\n");
+        if (bbmd6_address_match_self(addr)) {
+            /* ignore messages from my IPv6 address */
+        } else {
+            function_len = bvlc6_decode_forwarded_address_resolution(
+                pdu, pdu_len, &vmac_src, &vmac_target, &bip6_address);
+            if (function_len) {
```

Comment:
and here, rather than assuming go into `if` block when function_len is not null or 0.

Reply:
Whereas in principles I agree with your comments, we try to adhere to the existent coding standard - this is why we use the idiomatic syntax throughout - just to avoid re-work whilst submitting to the upstream.